lenovo/thinkpad/x230/default.nix

```diff
@@ -0,0 +1,26 @@
+{ config, lib, pkgs, ... }:
+
+with lib; {
+  imports = [
+    ../.
+    ../../../common/cpu/intel
+    ../../../common/pc/laptop/hdd
+  ];
+
+  boot = {
+    kernelModules = [
+      "kvm-intel"
+      "acpi_call"
+      "tpm-rng"
```

Comment:
`tpm-rng` is certainly a good idea, but our `rngd` doesn't use it: https://github.com/NixOS/nixpkgs/commit/64311899db7c455c5f07f7ff30c5a52c50b8fbae

Perhaps we should enable `services.tcsd`?

Reply:
actually it does. i tried running rngd -vvvv and the extra entropy source shows up only when tpm-rng is loaded

Reply:
Interesting!

Reply:
regarding tcsd, i had an extra import prepared ( `common/tpm` ) which would enable tcsd but i was unsure if there is actually any other way to use the tpm. i personally use the tcsd

Reply:
Seemingly, the correct way to use built-in TPM is to enable `tcsd`. I think it's a good idea to enable tcsd by default for all systems with TPM, but for now I think there is no need for a new `common/` profile, we'd have to check how it works on systems other than ThinkPads (e.g. Supermicro boards).

Reply:
i am unsure but it seems like the RNG of the tpm will only be used if `tpm-rng` is enabled even with tcsd enabled.
I will recheck however, the train to Lepizig will take another 3 hours anyway ;)

`rngd -vvv` with `tpm-rng` loaded:
```
Available entropy sources:
	Intel/AMD hardware rng
	DRNG
```

Reply:
I also think so, it seems both `tpm-rng` and `tcsd` is required to make use of this hardware. My point was that if just `tpm-rng` is loaded, we still need a daemon to read from TPM, because NixOS `services.rngd` module explicitly prohibits reading from TPM.

Reply:
Loading `tpm-rng` seems to be one way to have both `tcsd` and `rngd` get along together (according to some guy on the internet at least https://jpalczewski.github.io/en/posts/tpm-and-rngd/ )